File: src/Muflone/Messages/Events/IntegrationEvent.cs
﻿using Muflone.Core;
using Muflone.CustomTypes;
using System;

namespace Muflone.Messages.Events;

public abstract class IntegrationEvent : Event, IIntegrationEvent
{
	protected IntegrationEvent(IDomainId aggregateId, Guid correlationId, Account who) : base(aggregateId, correlationId, who)
	{
	}

	protected IntegrationEvent(IDomainId aggregateId, Account who) : base(aggregateId, who)
	{
	}

	protected IntegrationEvent(IDomainId aggregateId) : base(aggregateId)
	{
	}

	protected IntegrationEvent(IDomainId aggregateId, Guid correlationId) : base(aggregateId, correlationId)
	{
	}

	protected IntegrationEvent(IDomainId aggregateId, Guid correlationId, Account who, When when) : base(aggregateId, correlationId, who, when)
	{
	}
}

File: php/src/Muflone/Messages/Events/IntegrationEvent.php
<?php

declare(strict_types=1);

namespace Muflone\Messages\Events;

use Muflone\Core\IDomainId;
use Muflone\CustomTypes\Account;
use Muflone\CustomTypes\When;

abstract class IntegrationEvent extends Event implements IIntegrationEvent
{
    public function __construct(
        IDomainId $aggregateId,
        string $correlationId = '',
        ?Account $who = null,
        ?When $when = null
    ) {
        parent::__construct(
            $aggregateId,
            $correlationId,
            $who,
            $when
        );
    }
}
